src/components/EventMap.js

```diff
@@ -115,20 +128,113 @@ export default function(store){
         }
       },
 
-      createEmptyEventsDataSource() {
+      createEventsDataSource() {
         this.mapRef.addSource("events", {
           "type": "geojson",
-          "data": geoJsonHelpers.featureCollection([]),
+          // Depending on the order in which things load,
+          // this may be intitialized empty and may be ready to roll
+          "data": this.geojsonEvents,
           "cluster": true,
           "clusterMaxZoom": 8
         });
       },
 
+      // For all layers, we want the cursor to be a pointer on hover.
+      setCursorStyleOnHover() {
+        mapLayers.forEach(layer => {
+          this.mapRef.on('mouseenter', layer.id, (e) => {
+            this.mapRef.getCanvas().style.cursor = 'pointer';
+          });
+
+          this.mapRef.on('mouseleave', layer.id, (e) => {
+            this.mapRef.getCanvas().style.cursor = '';
+          });
+        });
+      },
+
+      openPopupsOnClick() {
+        const eventMap = this;
+
+        this.mapRef.on('click', 'unclustered-points', (e) => {
+          store.commit('eventSelected', eventId);
+
+          const eventId = e.features[0].properties.id;
+          const eventCoordinates = e.features[0].geometry.coordinates;
+
+          // Create a Vue instance _inside_ a mapbox Map instance
+          // _inside_ another Vue instance WHOAH. The point is
+          // to reuse the existing event card component.
+          const vm = new Vue({
```

Comment:
The length of this method is making me want to cry, maybe I'll split out this part at least :grin: